sharding-spring/sharding-jdbc-orchestration-spring/sharding-jdbc-orchestration-spring-boot-starter/src/test/java/org/apache/shardingsphere/shardingjdbc/orchestration/spring/boot/util/DataSourceUtilTest.java

```diff
@@ -119,4 +120,12 @@ public void assertDataSourceForLongValue() throws ReflectiveOperationException {
         assertThat(actual.getTimeBetweenEvictionRunsMillis(), is(16L));
         assertThat(actual.getMinEvictableIdleTimeMillis(), is(4000L));
     }
+
+    @Test
+    public void assertDataSourceForCollectionValue() throws ReflectiveOperationException {
+        Map<String, Object> dataSourceProperties = new HashMap<>(2, 1);
+        dataSourceProperties.put("connectionInitSqls", "[set names utf8mb4]");
```

Comment:
I think you should investigate how to config connectionInitSqls in raw spring-boot, then definition the list format which ShardingSphere should parsed.

Reply:
it seems "[]" will be parsed to array when using .yaml file, it's is better to support "," only.

Reply:
I have make it support "," only